plugin/contrib/mesos/pkg/executor/executor_test.go

```diff
@@ -195,3 +207,163 @@ func TestSuicide_WithTasks(t *testing.T) {
+}
+
+func TestExecutorReregister(t *testing.T) {
+	flag.Lookup("v").Value.Set(fmt.Sprint(*test_v))
+
+	mockDriver := MockExecutorDriver{}
+	executor := NewTestKubernetesExecutor()
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+
+	assert.Equal(t, executor.isConnected(), true, "executor should be connected")
+	mockDriver.AssertExpectations(t)
+}
+
+func TestExecutorDisconnect(t *testing.T) {
+	flag.Lookup("v").Value.Set(fmt.Sprint(*test_v))
+
+	mockDriver := MockExecutorDriver{}
+	executor := NewTestKubernetesExecutor()
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+	executor.Disconnected(mockDriver)
+
+	assert.Equal(t, executor.isConnected(), false, "executor should not be connected after Disconnected")
+	mockDriver.AssertExpectations(t)
+}
+
+func TestExecutorLaunchAndKillTask(t *testing.T) {
+	flag.Lookup("v").Value.Set(fmt.Sprint(*test_v))
+
+	mockDriver := MockExecutorDriver{}
+	config := Config{
+		Docker:    dockertools.ConnectToDockerOrDie("fake://"),
+		Updates:   make(chan interface{}, 1024),
+		APIClient: client.NewOrDie(&client.Config{Host: "fakehost", Version: testapi.Version()}),
+	}
+	executor := New(config)
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+
+	pod := newTestPod(1)
+	podTask, err := podtask.New(api.NewDefaultContext(), "",
+		*pod, &mesosproto.ExecutorInfo{})
+	assert.Equal(t, err, nil, "must be able to create a task from a pod")
+
+	taskInfo := podTask.BuildTaskInfo()
+	data, err := api.Codec.Encode(pod)
+	assert.Equal(t, err, nil, "must be able to encode a pod's spec data")
+	taskInfo.Data = data
+
+	mockDriver.On("SendStatusUpdate", mock.AnythingOfType("*mesosproto.TaskStatus")).Return(mesosproto.Status_DRIVER_RUNNING, nil)
+	executor.LaunchTask(mockDriver, taskInfo)
+	assert.Equal(t, len(executor.tasks), 1, "executor must be able to create a task")
+
+	mockDriver.On("SendStatusUpdate", mock.AnythingOfType("*mesosproto.TaskStatus")).Return(mesosproto.Status_DRIVER_RUNNING, nil)
+	executor.KillTask(mockDriver, taskInfo.TaskId)
+	assert.Equal(t, len(executor.tasks), 0, "executor must be able to kill a created task")
+	mockDriver.AssertExpectations(t)
+}
+
+func TestExecutorFrameworkMessage(t *testing.T) {
+	flag.Lookup("v").Value.Set(fmt.Sprint(*test_v))
+
+	mockDriver := MockExecutorDriver{}
+	executor := NewTestKubernetesExecutor()
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+	executor.FrameworkMessage(mockDriver, "test framework message")
+
+	// TODO(tyler) make this test more meaningful
+	mockDriver.AssertExpectations(t)
+}
+
+// Create a pod with a given index, requiring one port
+func newTestPod(i int) *api.Pod {
+	name := fmt.Sprintf("pod%d", i)
+	return &api.Pod{
+		TypeMeta: api.TypeMeta{APIVersion: testapi.Version()},
+		ObjectMeta: api.ObjectMeta{
+			Name:      name,
+			Namespace: "default",
+			SelfLink:  fmt.Sprintf("http://1.2.3.4/api/v1beta1/pods/%v", i),
```

Comment:
I seem to remember that k8s itself does multiple unit-test runs, one for each api. I haven't looked into how they do it tho. It might be worth looking into, to make sure we support them all.